### PR TITLE
fix(core/func): GetSpeedRate + SetSpeedRate + ifdefs

### PR DESCRIPTION
**Pull request by svenbledt**
Fixxed: SetSpeed and GetSpeed Function
Added: SetSpeedRate and GetSpeedRate
Removed: Unnecessary Ifdef's

Foe recognized on the azerothcore mod-eluna repo that the SetSpeed function is using the wrong SetSpeedRate member of unit.

This PR changes that and seperates the functions from each other.

Also we decided to remove the ifdefs that are not needed as the module is only for azerothcore and not for Trinty or Mangos.

You can test this PR by using the functions in a script and check the impact ;) 

Thanks for your time!

### CODE REVIEW

**Comment by 55Honey** [comment]
Thanks for the PR!

**Comment by r-o-b-o-t-o** [comment]
Thanks for your PR!
It would have been nice if the `ifdefs` were removed in a separate PR, it would have improved readability and traceability 😛

It looks like you're not using `forced` in `SetSpeedRate`? If that was a leftover from copy-pasting the other function, please remove the variable.

**Comment by 55Honey** [comment]
@svenbledt build is failing. Want to adjust?

**Comment by svenbledt** [comment]
> @svenbledt build is failing. Want to adjust?

i sadly dont know why build is failing :D worked fine on my end :o

File: src/LuaEngine/UnitMethods.h [hunk 1]
```diff
@@ -1561,11 +1349,44 @@ namespace LuaUnit
         (void)forced; // ensure that the variable is referenced in order to pass compiler checks
         if (type >= MAX_MOVE_TYPE)
             return luaL_argerror(L, 2, "valid UnitMoveType expected");
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/UnitMethods.h:1351]:
```suggestion
        if (type >= MAX_MOVE_TYPE)
        {
            return luaL_argerror(L, 2, "valid UnitMoveType expected");
        }
```

File: src/LuaEngine/UnitMethods.h [hunk 2]
```diff
@@ -1391,11 +1154,40 @@ namespace LuaUnit
...
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/UnitMethods.h:1155]:
```suggestion
        if (type >= MAX_MOVE_TYPE)
        {
            return luaL_argerror(L, 2, "valid UnitMoveType expected");
        }
```

**Review by r-o-b-o-t-o — CHANGES_REQUESTED** [review]
The CI should work again now.
Sorry I forgot to mention this earlier, could you please add a change to the README file to document your additions? See an [example here](https://github.com/azerothcore/mod-eluna/commit/7c6e6dbbd0f1ed8d723bede90d42970fbdb4725d)

**Review by svenbledt — COMMENTED** [review]
Thanks mate

File: README.md [hunk 3]
```diff
@@ -114,6 +114,8 @@ Eluna API for AC:
 - Added `Unit:GetUnitFlagsTwo()`: https://github.com/azerothcore/mod-eluna/pull/137
 - Added `Unit:SetUnitFlags(flags)`: https://github.com/azerothcore/mod-eluna/pull/137
 - Added `Unit:SetUnitFlagsTwo(flags)`: https://github.com/azerothcore/mod-eluna/pull/137
+- Added `Unit:SetSpeedRate(UnitMoveType, speed)`: https://github.com/azerothcore/mod-eluna/pull/155
```
Comment (r-o-b-o-t-o) [inline README.md:117]:
```suggestion
- Added `Unit:SetSpeedRate(unitMoveType, speed)`: https://github.com/azerothcore/mod-eluna/pull/155
```

**Pull request merged**